### PR TITLE
Add application configuration and installation files.

### PR DESCRIPTION
**Pull request by horakf**
The features this pull request implements are focused on project initialization and consists of these files:
- package installation files (setup.py, setup.cfg)
- project configuration file (pyproject.toml)
- pre-commit configuration (.pre-commit-config.yaml)
- documentation files (README.md, LICENSE)
- gitignore file (.gitignore)
- application directory structure ( __init__.py files)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+platforms = unix, linux, osx, cygwin, win32
+classifiers =
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3 :: Only
+    Programming Language :: Python :: 3.10
+
+[options]
+packages = data_clusterer
+install_requires =
+    numpy>=2.0.0
```
Comment (tynaneu) [inline setup.cfg:19]: it will be better to add the dependencies as they are needed (not in initial commits)

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    scikit-learn>=1.5.0
+    dependency-injector>=4.41.0
+python_requires = >=3.10.5
+package_dir =
+    =src
+zip_safe = no
+
+[options.extras_require]
+testing =
```
Comment (tynaneu) [inline setup.cfg:28]: pytest can be added in commit that adds pytest to pipeline/code

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    pytest>=8.2.2
+    pytest-cov>=5.0.0
+    mypy>=1.10.0
```
Comment (tynaneu) [inline setup.cfg:31]: hooks should also be installed/added when initialization pre-commit

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+requires = ["setuptools>=58.1.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.pytest.ini_options]
+addopts = "--cov=data_clusterer"
+testpaths = [
+    "src/tests",
+]
+
+[tool.mypy]
```
Comment (tynaneu) [inline pyproject.toml:11]: please add configuration of pre-commit hooks together with pre-commit settings (in one commit)

**Review by tynaneu — COMMENTED** [review]
It looks very nice. One nitpick, it is better to order changes to commits as they work together not according to files. For example, one commit should contain everything needed for pre-commit setup (among files).

**Comment by horakf** [comment]
I made the suggested changes and added CI workflow into this pull request. The tests dont pass because there is nothing to test. Should I provide some placeholder files, which will be deleted later, so the workflow passes or is it ok to keep it like that?

**Comment by tynaneu** [comment]
> I made the suggested changes and added CI workflow into this pull request. The tests dont pass because there is nothing to test. Should I provide some placeholder files, which will be deleted later, so the workflow passes or is it ok to keep it like that?

Yes, add some void test that will do nothing.

**Pull request merged**